src/EscrowSupplierNFT.sol

```diff
@@ -258,12 +276,16 @@ contract EscrowSupplierNFT is IEscrowSupplierNFT, BaseNFT {
      * @param loanId The associated loan ID
      * @return escrowId The ID of the created escrow
      */
-    function startEscrow(uint offerId, uint escrowed, uint fees, uint loanId)
+    function startEscrow(uint offerId, uint escrowed, uint fees, uint loanId, uint duration)
         external
         returns (uint escrowId)
     {
+        // Check if duration is within the offer's max duration
+        Offer memory offer = getOffer(offerId);
+        require(duration <= offer.maxDuration, "escrow: duration exceeds offer's max duration");
```

Comment:
med: this check needs to be in the validations in `_startEscrow()` instead. Currently it's not checked during `switchEscrow`. In this version of the code (with the ability to udpate maxDuration) it means a roll can ignore a new maxDuration value. If modifying duration is removed, in a future version of the code - if `switchEscrow` takes a `newDuration` as recommended (instead of using previous value) - it means that switchEscrow won't check the newDuration vs. maxDuration.